src/_pytest/_code/source.py

```diff
@@ -149,6 +149,9 @@ def get_statement_startend2(lineno: int, node: ast.AST) -> Tuple[int, Optional[i
     values: List[int] = []
     for x in ast.walk(node):
         if isinstance(x, (ast.stmt, ast.ExceptHandler)):
+            if isinstance(x, (ast.ClassDef, ast.FunctionDef, ast.AsyncFunctionDef)):
```

Comment:
I would add a comment above this line:

```
Before Python 3.8, the lineno of a decorated class or function pointed at the decorator.
Since Python 3.8, the lineno points to the class/def, so need to include the decorators.
```

Also, I figure a version check is not needed, because on Python<3.8 this would essentially just be a no-op (adding the same values twice), so it's not a problem.

Reply:
@bluetech Thank you, I added the those comment

Reply:
@bluetech Thank you, I added the those comment